Source/CDLCSegment.m

```diff
@@ -37,7 +37,7 @@ - (id)initWithDataCursor:(CDMachOFileDataCursor *)cursor;
         _segmentCommand.cmdsize = [cursor readInt32];
         
         _name = [cursor readStringOfLength:16 encoding:NSASCIIStringEncoding];
-        memcpy(_segmentCommand.segname, [_name UTF8String], sizeof(_segmentCommand.segname));
+        strncpy(_segmentCommand.segname, [_name UTF8String], sizeof(_segmentCommand.segname) - 1);
```

Comment:
If the original bug was that `memcpy` doesn't null-terminate, then the bug is still there: `strncpy` won't null-terminate either. Strongly consider using `snprintf` instead.